### ISSUE TITLE
Content.Load<Song> FileNotFoundException

### ISSUE THREAD

**Issue by MagellanicGames** [MonoGame/MonoGame#6075]
I posted on the MonoGame community about what could (maybe) be a bug on loading Song files from the content manager. 

Link to post here: http://community.monogame.net/t/content-load-song-causing-exception-looking-in-wrong-path/9987/2

The load function for "Song" is looking in the root of my drive for the sound file, instead of root of the "Content" directory.  This only happens with the project on my S: drive.  If I move the project to the C: drive there is no issue, it loads the Song just like the shader and image files.
The "Build Action" is set to "Build" and is using the "Ogg Importer - MonoGame".  Processor is set to "Song - MonoGame"

I am using the default cross-platform template with MonoGame 3.6.0.1625 on Windows 7 x64 with Visual Studio 2015 Update 3.   Code and exception output are below.  Apologies if I have omitted something important.

The constructor code of the class that inherites from the  "Game" class.
```
 public Main()
        {
            mGraphics = new GraphicsDeviceManager(this);
            mGraphics.PreferredBackBufferWidth = Dim.VIEW_RES_X;
            mGraphics.PreferredBackBufferHeight = Dim.VIEW_RES_Y;
            Content.RootDirectory = "Content";
            mCamera = new Camera(Dim.RES_X, Dim.RES_Y);
            mCullModeNone.CullMode = CullMode.None;
        }
```
Method for loading content:
 ```
protected override void LoadContent()
        {
            mSpriteSheet = Content.Load<Texture2D>("level-tiles");

            mShader = Content.Load<Effect>("Ambient");
        
            mDrawable = new Drawable(mSpriteSheet);
        
            mSong = Content.Load<Song>("bg_main_theme");
            MediaPlayer.Play(mSong);
            MediaPlayer.IsRepeating = true;
        
        }

```
The Exception:

> System.IO.FileNotFoundException was unhandled
>   FileName=S:\bg_main_theme.ogg
>   HResult=-2147024894
>   Message=Could not find file 'S:\bg_main_theme.ogg'.
>   Source=mscorlib
>   StackTrace:
>        at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
>        at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
>        at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
>        at Microsoft.Xna.Framework.Audio.OggStream.Open(Boolean precache)
>        at Microsoft.Xna.Framework.Audio.OggStream.Prepare()
>        at Microsoft.Xna.Framework.Media.Song.PlatformInitialize(String fileName)
>        at Microsoft.Xna.Framework.Media.Song..ctor(String fileName, Int32 durationMS)
>        at Microsoft.Xna.Framework.Content.SongReader.Read(ContentReader input, Song existingInstance)
>        at Microsoft.Xna.Framework.Content.ContentTypeReader`1.Read(ContentReader input, Object existingInstance)
>        at Microsoft.Xna.Framework.Content.ContentReader.InnerReadObject[T](T existingInstance)
>        at Microsoft.Xna.Framework.Content.ContentReader.ReadObject[T]()
>        at Microsoft.Xna.Framework.Content.ContentReader.ReadAsset[T]()
>        at Microsoft.Xna.Framework.Content.ContentManager.ReadAsset[T](String assetName, Action`1 recordDisposableObject)
>        at Microsoft.Xna.Framework.Content.ContentManager.Load[T](String assetName)
>        at BlockedCSMG.Main.LoadContent() in S:\c# Projects\BlockedCSMG\BlockedCSMG\Main.cs:line 76
>        at Microsoft.Xna.Framework.Game.Initialize()
>        at BlockedCSMG.Main.Initialize() in S:\c# Projects\BlockedCSMG\BlockedCSMG\Main.cs:line 44
>        at Microsoft.Xna.Framework.Game.DoInitialize()
>        at Microsoft.Xna.Framework.Game.Run(GameRunBehavior runBehavior)
>        at BlockedCSMG.Program.Main() in S:\c# Projects\BlockedCSMG\BlockedCSMG\Program.cs:line 17
>        at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
>        at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
>        at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
>        at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
>        at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
>        at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
>        at System.Threading.ThreadHelper.ThreadStart()
>   InnerException:

**Comment by KonajuGames** [comment]
Just a thought.  Have you tried it in a folder without a '#' in a folder name?  That's an odd character to use in a folder name, and maybe it is causing issues?

The other thing to look at is the contents of the XNB file, to see what the relative path to the .ogg file is.

**Comment by MagellanicGames** [comment]
I just changed the folder from "c# projects" to "cs projects" and it worked with no problems.  Hit the nail on the head 👍

**Comment by KonajuGames** [comment]
Thanks for confirming it.  Closing the issue.

**Comment by Jjagg** [comment]
Shouldn't we be able to handle that?

**Comment by KonajuGames** [comment]
My guess is the resolving of relative paths using the `Uri` class doesn't handle '#' characters.

**Comment by KonajuGames** [comment]
`ResolveRelativePath()` in Utilities/FileHelpers.cs(40)

**Comment by Jjagg** [comment]
There's a test that checks that `#` is handled properly by `ResolveRelativePath()`: https://github.com/MonoGame/MonoGame/pull/5583/files#diff-f52f2dd9d8c53d0ea21eb6591bbc4de2R45 
Maybe it's already wrong in the `.xnb` file.

**Comment by MagellanicGames** [comment]
I thought I might add, that even though '#' is an odd character to use; shouldn't it still handle it as the shader file and image file get resolved with no problems.  Their paths include the '#' character.

Sorry I didn't state that earlier, I thought I'd mentioned it here but I only mentioned it on the community forum.

**Comment by Jjagg** [comment]
Yes, this is probably an issue in the Song Processor or Writer and it should be fixed.

**Comment by NoFr1ends** [comment]
Happens still under MonoGame 3.6.
Should be something done here?

**Comment by cra0zy** [comment]
This is fixed in develop branch (3.7), not 3.6.
